src/main/java/org/broadinstitute/hellbender/tools/spark/sv/SVFastqUtils.java

```diff
@@ -4,49 +4,245 @@
 import com.esotericsoftware.kryo.Kryo;
 import com.esotericsoftware.kryo.io.Input;
 import com.esotericsoftware.kryo.io.Output;
+import htsjdk.samtools.Cigar;
 import htsjdk.samtools.SAMUtils;
+import htsjdk.samtools.TextCigarCodec;
+import htsjdk.samtools.util.Locatable;
+import htsjdk.samtools.util.SequenceUtil;
 import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.BaseUtils;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.Utils;
 import org.broadinstitute.hellbender.utils.fermi.FermiLiteAssembler;
 import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
 import org.broadinstitute.hellbender.utils.read.GATKRead;
 
 import java.io.*;
-import java.util.ArrayList;
-import java.util.Iterator;
-import java.util.List;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 /**
  * Memory-economical utilities for producing a FASTQ file.
  */
 public class SVFastqUtils {
 
+    public static final char HEADER_PREFIX_CHR = '@';
+    public static final char FRAGMENT_NUMBER_SEPARATOR_CHR = '/';
+    public static final char HEADER_FIELD_SEPARATOR_CHR = '\t';
+    public static final String HEADER_FIELD_SEPARATOR_REGEXP = "\\t"; // make sure its consistent with {@link #HEADER_FIELD_SEPARATOR_CHR}
+    public static final char HEADER_FIELD_EQUAL_CHR = '=';
+    private static final char HEADER_FIELD_ARRAY_SEPARATOR_CHR = ';';
+    private static final char LINE_SEPARATOR_CHR = '+';
+    private static final String MAPPING_FIELD_NAME = "mapping";
+    private static final String UNMAPPED_STR = "unmapped";
+
+    private static final String FRAGMENT_NUMBER_SEPARATOR_STR = "" + FRAGMENT_NUMBER_SEPARATOR_CHR;
+    private static final String HEADER_FIELD_SEPARATOR_STR = "" + HEADER_FIELD_SEPARATOR_CHR;
+    private static final String MAPPING_FIELD_EQUAL_TO = MAPPING_FIELD_NAME + HEADER_FIELD_EQUAL_CHR;
+    private static final String UNMAPPED_DESCRIPTION_STR = MAPPING_FIELD_EQUAL_TO + UNMAPPED_STR;
+    private static final String HEADER_FIELD_ARRAY_SEPARATOR_STR = "" + HEADER_FIELD_ARRAY_SEPARATOR_CHR;
+    private static final Pattern MAPPING_DESCRIPTION_PATTERN = Pattern.compile("^" + MAPPING_FIELD_EQUAL_TO +
+            "(.+):(\\d+);(" + Strand.PATTERN.pattern() + ");(.+)$");
+    private static final Pattern FASTQ_READ_HEADER_PATTERN = Pattern.compile("^" + HEADER_PREFIX_CHR +
+            "([^" + HEADER_FIELD_SEPARATOR_REGEXP +  "]+)(" + HEADER_FIELD_SEPARATOR_REGEXP + "(.*))?$");
+
+    // TODO htsjdk has it own Strand annotation enum. These classes could be merged if that Strand would me updated
+    // TODO so that one can get the enum constant char encoding; currently one can only do the transformation the other way.
```

Comment:
Also I guess you mention ```TemplateEnd``` here in relation to /1 /2... outputting rather than the strand + or -, which is what is addressed by the ```enum Strand``` this **TODO** is referring to.